cmd/packaging.go

```diff
@@ -0,0 +1,472 @@
+		if _, err := desktopFile.WriteString(line + "\n"); err != nil {
+			fmt.Printf("hover: Could not write %s.desktop: %v\n", projectName, err)
+			os.Exit(1)
+		}
+	}
+	err = desktopFile.Close()
+	if err != nil {
+		fmt.Printf("hover: Could not close %s.desktop: %v", projectName, err)
+		os.Exit(1)
+	}
+
+	gitignoreFilePath, err := filepath.Abs(filepath.Join(snapDirectoryPath, ".gitignore"))
+	if err != nil {
+		fmt.Printf("hover: Failed to resolve absolute path for .gitignore file %s: %v\n", gitignoreFilePath, err)
+		os.Exit(1)
+	}
+	gitignoreFile, err := os.Create(gitignoreFilePath)
+	if err != nil {
+		fmt.Printf("hover: Failed to create .gitignore file %s: %v\n", gitignoreFilePath, err)
+		os.Exit(1)
+	}
+	gitignoreFileContent := []string{
+		"assets",
+		"build",
+		"*.snap",
+	}
+
+	for _, line := range gitignoreFileContent {
+		if _, err := gitignoreFile.WriteString(line + "\n"); err != nil {
+			fmt.Printf("hover: Could not write .gitignore: %v\n", err)
+			os.Exit(1)
+		}
+	}
+	err = gitignoreFile.Close()
+	if err != nil {
+		fmt.Printf("hover: Could not close .gitignore: %v", err)
+		os.Exit(1)
+	}
+
+	fmt.Printf("hover: You now can package the snap using `hover build %s`\n", packagingFormat)
+}
+
+func buildLinuxSnap(projectName string) {
+	packagingFormat := "linux-snap"
+	assertCorrectOS(packagingFormat)
+	snapDirectoryPath := filepath.Join(packagingFormatPath(packagingFormat))
+	snapcraftBin, err := exec.LookPath("snapcraft")
+	if err != nil {
+		fmt.Println("hover: Failed to lookup `snapcraft` executable. Please install snapcraft.\nhttps://tutorials.ubuntu.com/tutorial/create-your-first-snap#1")
+		os.Exit(1)
+	}
+	fmt.Println("hover: Packaging snap...")
+
+	err = copy.Copy(filepath.Join(buildPath, "assets"), filepath.Join(packagingFormatPath(packagingFormat), "assets"))
+	if err != nil {
+		fmt.Printf("hover: Could not copy assets folder: %v", err)
+		os.Exit(1)
+	}
+	err = copy.Copy(outputDirectoryPath("linux"), filepath.Join(packagingFormatPath(packagingFormat), "build"))
+	if err != nil {
+		fmt.Printf("hover: Could not copy build folder: %v", err)
+		os.Exit(1)
+	}
+
+	cmdBuildSnap := exec.Command(snapcraftBin)
+	cmdBuildSnap.Dir = filepath.Join(snapDirectoryPath)
+	cmdBuildSnap.Stdout = os.Stdout
+	cmdBuildSnap.Stderr = os.Stderr
+	cmdBuildSnap.Stdin = os.Stdin
+	err = cmdBuildSnap.Run()
+	if err != nil {
+		fmt.Printf("hover: Failed to package snap: %v", err)
+		os.Exit(1)
+	}
+	outputFilePath := filepath.Join(outputDirectoryPath("linux-snap"), escapedProjectName(projectName)+"_"+runtime.GOARCH+".snap")
+	err = os.Rename(filepath.Join(snapDirectoryPath, escapedProjectName(projectName)+"_"+assertInFlutterProject().Version+"_"+runtime.GOARCH+".snap"), outputFilePath)
+	if err != nil {
+		fmt.Printf("hover: Could not move snap file: %v\n", err)
+		os.Exit(1)
+	}
+}
+
+func initLinuxDeb(projectName string) {
+	packagingFormat := "linux-deb"
+	assertCorrectOS(packagingFormat)
+	if assertInFlutterProject().Author == "" {
```

Comment:
The author field isn't generated by `flutter create XXX`.
If the author field isn't found in the pubspec.yaml, we can fallback to [$USER](https://golang.org/pkg/os/user/#Current) instead of failing here.

Reply:
I agree with you, but $USER is only available on linux and darwin and would therefore be inconsistent on the windows.

Reply:
I was referring to https://golang.org/pkg/os/user/#Current :wink:

Reply:
That only works on unix systems I think.

Reply:
Unless otherwise noted, os-package functions should work on all systems, even Plan9. Looking at the docs, it mentions that the field Name is the Display Name (John Doe), rather than login name (john). Perhaps we want to use the Display name? (I'm not sure how it's used)

Reply:
The login name should be used, because the field in the control file uses names without spaces and refernces usernames from snapcraft